### PR TITLE
feat: adding tag table and implementing tag filtering

### PR DESCRIPTION
**Pull request by mcgingras**
started spiking a branch for adding tags and filtering by tags. Will go into more detail as I flesh this out, just a spike for now.
We need to think about how we want to keep track of state for the non user-assigned tag types. Maybe it makes sense to not have those be db tag types but instead be filter that we apply behind the scenes. For example hot might not be a tag at the db level that we filter on, but we instead have some logic in the ideaService such that when a tag of hot is applied we are really adding a filter for where tag.votes > 50 and tag.created_at < 1 week or something

### CODE REVIEW

File: packages/nouns-api/src/graphql/resolvers/PropLotListResolvers.ts [hunk 1]
```diff
@@ -135,6 +138,14 @@ const resolvers: IResolvers = {
         type: FilterType.MultiSelect,
         label: 'Tags',
         options: [
+          {
```
Comment (ThatTobMate) [inline packages/nouns-api/src/graphql/resolvers/PropLotListResolvers.ts:141]: I added a comment in the code a few lines above this but i'd probably load the tags from the DB here. then iterate over them and build out these options using that data.

I'll comment it here but to support this i'd maybe add a displayName property to your tag model and you can load those in here for the label. Then i'd replace the hardcoded `'REQUEST'` values with the TagType property.

Would endup looking something like

```
 const tags: Tag[] = prisma.tags.findMany(...)
 
 const filterOptions = tags.map(tag => {
  return {
     id: `${FILTER_IDS.TAG}-${tag.type}`
     label: tag.displayName,
     value: buildFilterParam(FILTER_IDS.TAG, tag.type),
     selected: Boolean(
         root.tagParams?.includes(buildFilterParam(FILTER_IDS.TAG, tag.type)),
     ),
  }
})

const tagFilter: PropLotFilter = {
 ....
 options: filterOptions,
}
```

Comment (mcgingras) [reply packages/nouns-api/src/graphql/resolvers/PropLotListResolvers.ts:141]: Yea, I like the way this looks.

Comment (ThatTobMate) [reply packages/nouns-api/src/graphql/resolvers/PropLotListResolvers.ts]: On the point about tags like Hot I'd probably create a map for these "virtual tags" and then load them in here. Something like this?

```
// load + build DB tags

const virtualTags = {
  HOT: {
    type: HOT,
    displayName: "Hot",
    filterFn: (ideas) => { ... logic to filter ideas }, // not sure about this property
  },
  ...
}; // Maybe store these in a standalone file and import them here - virtualTags.ts?

 const virtualFilterOptions = Object.keys(virtualTags).map(vT => {
  return {
     id: `${FILTER_IDS.TAG}-${virtualTags[vT].type}`
     label: tag.displayName,
     value: buildFilterParam(FILTER_IDS.TAG, virtualTags[vT].type),
     selected: Boolean(
         root.tagParams?.includes(buildFilterParam(FILTER_IDS.TAG, virtualTags[vT].type)),
     ),
  }
})


const tagFilter: PropLotFilter = {
 ....
 options: [...filterOptions, ...virtualFilterOptions],
}
```

**Comment by ThatTobMate** [comment]
I've merged the other PRs into this one https://github.com/ThatTobMate/lilnouns-monorepo/pull/29 so there aren't loads of branches going on, you may need to rebase this one now though to sync them up and fix conflicts.

**Pull request closed without merging**